web/js/components/timeline/timeline-data/data-item-container.js

```diff
@@ -109,13 +120,22 @@ class DataItemContainer extends Component {
       dataDateRanges,
     } = this.state;
 
+    // layer options
     const {
-      endDate,
+      futureTime,
       id,
       startDate,
       visible,
     } = layer;
+    let {
```

Comment:
Modifying the layer's `endDate` property here in the render function of a deeply nested component seems risky. Since we're just modifying a date slightly, this probably won't introduce any bugs but as a general practice I think it's best avoided in cases like this where other parts of the app are reading this property.  If you were, for example, debugging something elsewhere in the app and looking at the `endDate` property and you noticed it didn't match up with what's in the `wv.json`, you might have a tough time tracking down where this is being modified.

Additionally, calling `getFutureLayerEndDate` on every render seems unnecessary since it shouldn't be changing once the app loads.  It looks like you are having to call this in several other places throughout the app as well which leads me to think maybe it's best to find a place to call it once and be done.

Perhaps a solution to both is to handle all `endDate` modifications in a central place early on in the app lifecycle?  I had to do something like this to adjust `startDates` for geostationary layers and did it [here in `main.js`](https://github.com/nasa-gibs/worldview/blob/92d27c795321ae13a794bb0d9027dad05aee0ac6/web/js/main.js#L174).

Reply:
Yeah changing the `endDate` there is dangerous. I'll get that change up.

For the `getFutureLayerEndDate` I was under the initial impression that the future end dates needed to be dynamic and keep updating if a user left their browser open, but on second thought based on what you said, adjusting the `endDate` early in the app vs repeatedly makes more sense for the use case as users can always refresh.